index.js

```diff
@@ -274,10 +274,10 @@ function parse (uri, opts) {
         parsed.host = unescape(parsed.host)
       }
       if (parsed.path !== undefined && parsed.path.length) {
-        parsed.path = escape(parsed.path)
+        parsed.path = encodeURI(parsed.path)
       }
       if (parsed.fragment !== undefined && parsed.fragment.length) {
-        parsed.fragment = escape(parsed.fragment)
+        parsed.fragment = encodeURI(decodeURI(parsed.fragment))
```

Comment:
I miss the [`escape`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape?retiredLocale=it) usage.
There is any reason to prefer it over `encodeURI` and `decodeURI`?

This fix could impact the performance.. need to try it

Reply:
Iirc i switched to escape because it was faster. I'll have to run some benchmarks. But at the moment i switched i cleaned up a lot the parse flow. Going to check